autokeras/hypermodel/block.py

```diff
@@ -0,0 +1,599 @@
+                                                 default=False)
+        dropout_rate = self.dropout_rate or hp.Choice('dropout_rate',
+                                                      [0, 0.25, 0.5],
+                                                      default=0)
+
+        for i in range(num_layers):
+            units = hp.Choice(
+                'units_{i}'.format(i=i),
+                [16, 32, 64, 128, 256, 512, 1024],
+                default=32)
+            output_node = tf.keras.layers.Dense(units)(output_node)
+            if use_bn:
+                output_node = tf.keras.layers.BatchNormalization()(output_node)
+            output_node = tf.keras.layers.ReLU()(output_node)
+            output_node = tf.keras.layers.Dropout(dropout_rate)(output_node)
+        return output_node
+
+
+class RNNBlock(HyperBlock):
+    """An RNN HyperBlock.
+
+    # Arguments
+        return_sequences: Boolean. Whether to return the last output in the
+            output sequence, or the full sequence. Defaults to False.
+        bidirectional: Boolean. Bidirectional RNN. If left unspecified, it will be
+            tuned automatically.
+        num_layers: Int. The number of layers in RNN. If left unspecified, it will
+            be tuned automatically.
+        layer_type: Str. 'gru' or 'lstm'. If left unspecified, it will be tuned
+            automatically.
+    """
+
+    def __init__(self,
+                 return_sequences=False,
+                 bidirectional=None,
+                 num_layers=None,
+                 layer_type=None,
+                 **kwargs):
+        super().__init__(**kwargs)
+        self.return_sequences = return_sequences
+        self.bidirectional = bidirectional
+        self.num_layers = num_layers
+        self.layer_type = layer_type
+
+    def build(self, hp, inputs=None):
+        inputs = nest.flatten(inputs)
+        utils.validate_num_inputs(inputs, 1)
+        input_node = inputs[0]
+        shape = input_node.shape.as_list()
+        if len(shape) != 3:
+            raise ValueError(
+                'Expect the input tensor to have '
+                'at least 3 dimensions for rnn models, '
+                'but got {shape}'.format(shape=input_node.shape))
+
+        feature_size = shape[-1]
+        output_node = input_node
+
+        bidirectional = self.bidirectional or hp.Choice('bidirectional',
+                                                        [True, False],
+                                                        default=True)
+        layer_type = self.layer_type or hp.Choice('layer_type',
+                                                  ['gru', 'lstm'],
+                                                  default='lstm')
+        num_layers = self.num_layers or hp.Choice('num_layers',
+                                                  [1, 2, 3],
+                                                  default=2)
+        rnn_layers = {
+            'vanilla': tf.keras.layers.SimpleRNN,
```

Comment:
This is not included in the `layer_type` hp, and does not seem to be useful in general.